### ISSUE TITLE
Fatal: Not supported tflite opcode: SHAPE

### ISSUE THREAD

**Issue by juliowissing** [kendryte/nncase#177]
When trying to compile my model I get a "Not supported tflite opcode:SHAPE" error. However, I am not using the shape command when building my model. I only use reshape and flatten to manipulate the shape. My model has the following structure:
<div>
_________________________________________________________________<br/>
Layer (type)                 Output Shape              Param #<br/>
========================================<br/>
reshape (Reshape)            (None, 70, 20, 1)         0<br/>
_________________________________________________________________<br/>
conv2d (Conv2D)              (None, 26, 1, 2)          802<br/>
_________________________________________________________________<br/>
activation (Activation)      (None, 26, 1, 2)          0<br/>
_________________________________________________________________<br/>
batch_normalization (BatchNo (None, 26, 1, 2)          8<br/>
_________________________________________________________________<br/>
flatten (Flatten)            (None, 52)                0<br/>
_________________________________________________________________<br/>
dropout (Dropout)            (None, 52)                0<br/>
_________________________________________________________________<br/>
dense (Dense)                (None, 16)                848<br/>
_________________________________________________________________<br/>
activation_1 (Activation)    (None, 16)                0<br/>
_________________________________________________________________<br/>
dropout_1 (Dropout)          (None, 16)                0<br/>
_________________________________________________________________<br/>
dense_1 (Dense)              (None, 8)                 136<br/>
_________________________________________________________________<br/>
activation_2 (Activation)    (None, 8)                 0<br/>
_________________________________________________________________<br/>
dense_2 (Dense)              (None, 5)                 45<br/>
========================================<br/>
Total params: 1,839<br/>
Trainable params: 1,835<br/>
Non-trainable params: 4<br/>
</div>

Anyone have an idea whats going on?

**Comment by juliowissing** [comment]
I resolved the issue by not using the keras layer "Reshape". Instead I went with a functional keras model and used the tf.reshape operation to ommit the layer, which does use the tf.shape operation internally in the process.
